dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/transform/strategy/DefaultTransformStrategy.java

```diff
@@ -0,0 +1,302 @@
+        addLanguage(contentlet, map, options);
+        addVersionProperties(contentlet, map, options);
+        addConstants(contentlet, map, options);
+        addBinaries(contentlet, map, options);
+        addCategories(contentlet, map, options, user);
+        return map;
+    }
+
+    /**
+     * Handle common properties found on all contentlets
+     * @param contentlet
+     * @param map
+     * @param options
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    private void addCommonProperties(final Contentlet contentlet, final Map<String, Object> map, final Set<TransformOptions> options)
+            throws DotSecurityException, DotDataException {
+        if(!options.contains(COMMON_PROPS)){
+            return;
+        }
+        final ContentType type = contentlet.getContentType();
+
+        map.put(IDENTIFIER_KEY, contentlet.getIdentifier());
+        map.put(INODE_KEY, contentlet.getInode());
+        map.put(TITTLE_KEY, contentlet.getTitle());
+        map.put(CONTENT_TYPE_KEY, type != null ? type.variable() : NOT_APPLICABLE);
+        map.put(BASE_TYPE_KEY, type != null ? type.baseType().name() : NOT_APPLICABLE);
+        map.put(LANGUAGEID_KEY, contentlet.getLanguageId());
+        final Optional<Field> titleImage = contentlet.getTitleImage();
+        final boolean hasTitleImage = titleImage.isPresent();
+        map.put(HAS_TITLE_IMAGE_KEY, hasTitleImage);
+        if(hasTitleImage) {
+           map.put(TITLE_IMAGE_KEY, titleImage.get().variable());
+        } else {
+           map.put(TITLE_IMAGE_KEY, TITLE_IMAGE_NOT_FOUND);
+        }
+        final Host host = toolBox.hostAPI.find(contentlet.getHost(), APILocator.systemUser(), true);
+        map.put(HOST_NAME, host != null ? host.getHostname() : NOT_APPLICABLE);
+        map.put(HOST_KEY, host != null ? host.getIdentifier() : NOT_APPLICABLE);
+
+        final String urlMap = toolBox.contentletAPI
+                .getUrlMapForContentlet(contentlet, toolBox.userAPI.getSystemUser(), true);
+        map.put(ESMappingConstants.URL_MAP, urlMap);
+
+        //We only calculate the fields if it is not already set
+        //However WebAssets (Pages, FileAssets) are forced to calculate it.
+        //To prevent any miscalculated urls.
+        if (!map.containsKey(URL_FIELD)) {
+            final String url = toolBox.contentHelper.getUrl(contentlet);
+            if (null != url) {
+                map.put(URL_FIELD, url);
+            }
+        }
+    }
+
+    /**
+     * if TransformOptions.USE_LANGUAGE_AS_MAP is set this will load all the language related properties as a single map entry
+     * @param contentlet
+     * @param map
+     * @param options
+     */
+    private void addLanguage(final Contentlet contentlet, final Map<String, Object> map,
+            final Set<TransformOptions> options) {
+        final Language language = toolBox.languageAPI.getLanguage(contentlet.getLanguageId());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5596402)